added string to multiple python file generation for skill library

akuten1298 opened the pull request:
### Describe the changes you have made:
Added functionality to convert from string into python code and store in a dict
### Reference any relevant issues (e.g. "Fixes #000"):

### Pre-Submission Checklist (optional but appreciated):

- [ ] I have included relevant documentation updates (stored in /docs)
- [ ] I have read `docs/CONTRIBUTING.md`
- [ ] I have read `docs/ROADMAP.md`

### OS Tests (optional but appreciated):

- [ ] Tested on Windows
- [ ] Tested on MacOS
- [ ] Tested on Linux

KillianLucas commented:
Incredible work Aakash! Will merge if you want to move this function into `interpreter/core/computer/terminal/languages/jupyter_language.py` — I put a function at the at there called `string_to_python`, if you populate it with your code we'll be one step closer.

I think it should be changed to just handle the conversion instead of the saving tho— basically ~ line 102, I think it should just return a dictionary in this format:

`{"function_name": "import whatever\ndef function_name(): ..."}`

Where the key is the function name and the value is the code. Then OI will handle saving it to the right place. Let me know if that makes sense!

akuten1298 commented:
Got it, sure. I've added that change and updated the PR, lmk if this is exactly what you are looking for. Attaching an image here to make it easier to check.

<img width="1508" alt="Screenshot 2024-02-04 at 7 24 58 AM" src="https://github.com/KillianLucas/open-interpreter/assets/51192400/1c9dce81-a193-4c3d-84fa-34b7b02fdc68">

KillianLucas commented:
Forgot to thank you for this after merging Aakash, brilliant work. Also we're still at the Airbnb if you're free tomorrow :)
